components/api/src/main/java/com/hotels/styx/api/client/Connection.java

```diff
@@ -68,6 +70,14 @@
      */
     <R> Observable<R> execute(Supplier<Observable<R>> operation);
 
+    /**
+     * Writes HTTP request to a remote peer in the context of this connection.
+     *
+     * @param request
+     * @return an observable that provides the response
+     */
+    Observable<HttpResponse> write(HttpRequest request);
+
```

Comment:
The `write` method should also appear in the `Connection` interface. As for the `Connection` interface, it probably needs type parameters in place of HttpRequest and HttpResponse.

Reply:
But this is a connection interface.

Reply:
IMO it makes sense. Looking at `Connection` interface there is nothing that is inherently specific to HTTP protocol. And it is just our particular implementation of `write` which is a function `HttpRequest -> Observable<HttpResponse>` and this could be expressed in `Connection` in terms of type parameters.

But then again this is not really a top priority at the moment. So let's leave this as it is for now.